packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -0,0 +1,77 @@
+import { Element } from "hast";
+import React from "react";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+//Make sure this uses cjs, not esm. See: https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/509
+import * as style from "react-syntax-highlighter/dist/cjs/styles/prism";
+import remarkGfm from "remark-gfm";
+import { Node, Parent } from "unist";
+import { visitParents } from "unist-util-visit-parents";
+
+// Adds `inline` property to `code` elements, to distinguish between inline and block code snippets.
+function rehypeInlineCodeProperty() {
+  return function (tree: Node) {
+    visitParents(tree, "element", function (node: Node, parents: Parent[]) {
+      const element = node as Element;
+      if (element.tagName === "code") {
+        const lastParent = parents[parents.length - 1] as Element;
+        if (!element.properties) element.properties = {};
+        //It should be a string, because it can't be a boolean.
+        element.properties.inline = String(
+          !(lastParent && lastParent.tagName === "pre")
+        );
+      }
+    });
+  };
+}
+
+type MarkdownViewerProps = {
+  md: string;
+};
+
+const MarkdownViewer: React.FC<MarkdownViewerProps> = ({ md }) => {
+  return (
+    <ReactMarkdown
+      className="prose prose-stone prose-sm"
+      rehypePlugins={[rehypeInlineCodeProperty]}
+      remarkPlugins={[remarkGfm]}
+      components={{
```

Comment:
Inline components here make me slightly nervous (creating React components inside other components is very discouraged).

OTOH, react-markdown docs use this style too, I hope this means that these are actually render callbacks and not real components, then it's fine.

Reply:
Yea, I was just following their examples here.